dts/bindings/mtd/cypress,fm24w256.yaml

```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2021 Caspar Friedrich <c.s.w.friedrich@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+description: 256-Kbit (32K × 8) Serial (I2C) F-RAM
+
+compatible: "cypress,fm24w256"
+
+include: [eeprom-base.yaml, i2c-device.yaml]
+
+properties:
+  wp-gpios:
+    type: phandle-array
+    required: false
+    description: |
+      Enable write protection on all addresses, active high.
+      The functionality is not supported by the driver. If the pin is set, the
+      driver ensures that the GPIO is configured correctly to not interfere with
+      normal operation.
```

Comment:
If we are to include this GPIO in the DTS and initialise it in the driver, I think we need to make the driver actively use it as well. Check out the AT2x EEPROM driver (`drivers/eeprom/eeprom_at2x.c`) and see how it utilises the optional WP pin for enabling/disabling writes.